OVERVIEW.md

```diff
@@ -0,0 +1,484 @@
+transmit number. If an observed `upkeepKey` occured at a block previous to the
+last transmit block number, the key will be filtered out of the observation.
+This ensure observations are not stale based on tracked internal state.
+
+```
+define [upkeepKey]
+define [blockedIDs]
+define [lastTransmitBlock]
+
+split (upkeepKey) into [blockNumber], [upkeepID]
+
+if (upkeepID) is in (blockedIDs) and (blockNumber) is before or equal to (lastTransmitBlock)
+    filter out (upkeepKey)
+```
+
+**Transmitting**
+
+Transmission confirmation is done on the `upkeepKey` directly. If there is a key
+in pending state that matches the `upkeepKey` exactly, the perform action has
+not yet been confirmed.
+
+```
+if (upkeepKey) is not in (pendingKeys)
+    return transmissionConfirmed
+else if (upkeepKey) is in (pendingKeys) and (confirmationState) is true
+    return transmissionConfirmed
+else
+    return transmissionNotConfirmed
+```
+
+**Logs**
+
+Logs are polled every second and passed through the following:
+
+Perform log entries indicate that a perform exists on chain in some capacity.
+The existance of an entry means that the transaction was broadcast by at least 
+one node. Reorgs can still happen causing performs to get moved to a later block
+or change to reorg logs. Higher minConfirmations setting reduces the chances of 
+this happening.
+
+We do two things upon receiving a perform log:
+
+- Mark the upkeep key responsible for the perform as 'transmitted', so that this node does not waste gas trying to transmit the same report again
+- Unblock the upkeep from idBlocks so that it can be observed and reported on again.
+
+```
+define [performLogs]
+define [configMinConfirmations]
+define [pendingKeys]
+define [blockedIDs]
+
+for each (performLogs)
+    define [upkeepKey]
+    define [performLogConfirmations]
+    define [performBlockNumber]
+
+    if (performLogConfirmations) >= (configMinConfirmations)
+        split (upkeepKey) into [blockNumber], [upkeepID]
+
+        if (upkeepKey) in (pendingKeys)
+            define [upkeepKeyConfirmationState]
+
+            if (upkeepKeyConfirmationState) is false
+                set (upkeepKey) in (pendingKeys) with expiration of [pendingKeyTimeout] and confirmed true
+
+                set (upkeepID) in (blockedIDs) with (blockNumber) and (performBlockNumber)
+            else
+                if (upkeepID) is in (blockedIDs) and
+                    (blockNumber) == [blockedIDBlockNumber] and 
+                    (performBlockNumber) != [blockedIDBlockNumber]
+
+                    set (upkeepID) in (blockedIDs) with (blockNumber) and (performBlockNumber)
+
+```
+
+It can happen that in between the time the report is generated and it gets
+confirmed on chain something changes and it becomes stale. Current scenarios are:
+
+- Another report for the upkeep is transmitted making this report stale
+- Reorg happens which changes the checkBlockHash making this reorged report as it was checked on a different chain
```

Comment:
I don't have a sufficiently precise understanding of what's going on here to assess whether a re-org can put the system in a bad state.